frame/support/src/storage/mod.rs

```diff
@@ -1413,6 +1542,52 @@ mod test {
 		});
 	}
 
+	#[test]
+	fn prefix_iterator_pagination() {
+		TestExternalities::default().execute_with(|| {
+			use crate::hash::Identity;
+			crate::generate_storage_alias! {
+				MyModule,
+				MyStorageMap => Map<(u64, Identity), u64>
+			}
+
+			MyStorageMap::insert(1, 10);
+			MyStorageMap::insert(2, 20);
+			MyStorageMap::insert(3, 30);
+			MyStorageMap::insert(4, 40);
+			MyStorageMap::insert(5, 50);
+			MyStorageMap::insert(6, 60);
+			MyStorageMap::insert(7, 70);
+			MyStorageMap::insert(8, 80);
+			MyStorageMap::insert(9, 90);
+			MyStorageMap::insert(10, 100);
+
+			let op = |(_, v)| v / 10;
+			let mut final_vec = vec![];
+			let mut iter = MyStorageMap::iter();
+		
+			let elem = iter.next().unwrap();
+			assert_eq!(elem, (1, 10));
+			final_vec.push(op(elem));
+
+			let elem = iter.next().unwrap();
+			assert_eq!(elem, (2, 20));
+			final_vec.push(op(elem));
+
+			let stored_key = iter.last_raw_key().to_owned();
+			let iter = MyStorageMap::iter_from(stored_key).map(op);
+			let mut remaining = iter.collect::<Vec<_>>();
+			remaining.sort();
+			assert_eq!(remaining.len(), 8);
+			assert_eq!(remaining, vec![3, 4, 5, 6, 7, 8, 9, 10]);
+
+			final_vec.extend_from_slice(&remaining);
+			final_vec.sort();
+
+			assert_eq!(final_vec, vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+		});
+	}
```

Comment:
The new doc comments say something along the lines of "Enumerate all elements in the map after a specified `starting_raw_key` in no particular", but the test here makes it look like we enumerate in the order of insertion.

So to clarify:
- If we are to iterate over a map single, non mutated map multiple times, will we get the same ordering each time?
- If we do not get the same ordering, what is the point of having a starting key? Seems like that key could be at any position in the ordering if ordering is variant, which does not seem useful